File: TesteCredito/Data/Domain/Creditos/Financiamento.cs
using Domain.Enums;

namespace Domain.Creditos
{
    public class Financiamento
    {
        public Guid Id { get; set; }
        public double Valor { get; set; }
        public TipoCredito Tipo { get; set; }
        public int QuantidadeParcelas { get; set; }

        public List<Parcela> Parcelas { get; set; }
    }
}

File: teste-credito-front/src/app/shared/models/financiamento.ts
export class Financiamento {
    public Valor!: number | null;
    public Tipo!: string | null;
    public QuantidadeParcelas!: number | null;
    public DataPrimeiraParcela!: Date | null;
    public ValorJuros!: number | null;
    public Status!: string | null;

    constructor(){
    }
}